modules/core/src/main/scala/fs2/kafka/consumer/MkConsumer.scala

```diff
@@ -30,7 +30,7 @@ object MkConsumer {
   implicit def mkConsumerForSync[F[_]](implicit F: Sync[F]): MkConsumer[F] =
     new MkConsumer[F] {
 
-      def apply[G[_]](settings: ConsumerSettings[G, ?, ?]): F[KafkaByteConsumer] = F.delay {
+      def apply[G[_]](settings: ConsumerSettings[G, ?, ?]): F[KafkaByteConsumer] = F.blocking {
```

Comment:
`WithConsumer` manages the allocation of a dedicated blocking context for the consumer instance provided by `MkConsumer`. We should probably update `WithConsumer` to ensure that the instance is constructed in that context, instead of using another blocking context here.

Reply:
Hmm, not entirely sure. Doesn't `WithConsumer` create a new blocking context for consumer to use? (unless a custom pool is provided)
```
    val blocking: Resource[F, Blocking[F]] = settings.customBlockingContext match {
      case None     => Blocking.singleThreaded[F]("fs2-kafka-consumer")
      case Some(ec) => Resource.pure(Blocking.fromExecutionContext(ec))
    }
```
And here we would be using default ce3 context to initialize the consumer .